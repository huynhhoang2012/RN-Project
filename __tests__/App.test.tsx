import React from 'react';
// import {create, act} from 'react-test-renderer';
import LoginScreen from '../src/screens/Login';
import {render, fireEvent} from '@testing-library/react-native';

// const tree = create(<LoginScreen />);

// test('snapshot', () => {
//   expect(tree).toMatchSnapshot();
// });

// test('button click', () => {
//   const button = tree.root.findByProps({testID: 'buttonLogin'}).props;
//   act(() => button.onPress());

//   const text = tree.root.findByProps({testID: 'testText'}).props;
//   expect(text.children).toEqual('admin');
// });

test('renders login screen correctly', () => {
  const {getByPlaceholderText, getByText} = render(<LoginScreen />);

  const usernameInput = getByPlaceholderText('Username');
  const passwordInput = getByPlaceholderText('Password');
  const loginButton = getByText('Login');

  expect(usernameInput).toBeTruthy();
  expect(passwordInput).toBeTruthy();
  expect(loginButton).toBeTruthy();
});

test('displays status login', () => {
  const {getByPlaceholderText, getByText} = render(<LoginScreen />);

  const usernameInput = getByPlaceholderText('Username');
  const passwordInput = getByPlaceholderText('Password');
  const loginButton = getByText('Login');

  fireEvent.changeText(usernameInput, 'admin');
  fireEvent.changeText(passwordInput, 'admin');
  fireEvent.press(loginButton);

  const successMessage = getByText('login successful');
  expect(successMessage).toBeTruthy();
});
