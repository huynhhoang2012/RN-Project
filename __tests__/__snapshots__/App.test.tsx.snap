// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "700",
          "marginBottom": 20,
        }
      }
      testID="testText"
    />
    <View>
      <Text>
        User name
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          {
            "borderColor": "#D9D9D9D9",
            "borderRadius": 6,
            "borderWidth": 1,
            "fontSize": 14,
            "fontWeight": "400",
            "marginBottom": 20,
            "marginTop": 4,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
        testID="userNameText"
        value=""
      />
    </View>
    <View>
      <Text>
        Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          {
            "borderColor": "#D9D9D9D9",
            "borderRadius": 6,
            "borderWidth": 1,
            "fontSize": 14,
            "fontWeight": "400",
            "marginBottom": 20,
            "marginTop": 4,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
        testID="passwordText"
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#213960",
          "borderRadius": 6,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
      testID="buttonLogin"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
